src/main.rs

```diff
@@ -17,32 +18,41 @@ impl WeatherResponse {
         let url = Url::parse(&*url)?;
         println!("{}", url);
         let res = reqwest::get(url).await?.json::<serde_json::Value>().await?;
-        //unwrap current field of res or handle error   
-        match res.get("current") {
-            Some(x) => {
-                match _x.get("temp_c") {  
-                    Some(y) => {
-                        println!("current field found");
-                        let weather_res = WeatherResponse {temp_c: y.as_f64().unwrap() as f32};
-                        Ok(weather_res)
-                    },
-                    None => {
-                        println!("temp_c field not found");
-                        let weather_res = WeatherResponse {temp_c: 0.0};
-                        Ok(weather_res)
-                    }
+        
+        let temp = res.get("current").and_then(|x| x.get("temp_c")).and_then(|x| x.as_f64());
+        match temp {
+            Some(x) => Ok(WeatherResponse{temp_c: x as f32, forecast: None}),
+            None => Err(ExitFailure::from(std::io::Error::new(std::io::ErrorKind::Other, "API Error")))
+        }
+    }
+    async fn get_today(location: &String, api_key: &String) -> Result<Self, ExitFailure> {
+        let url = format!(
+            "http://api.weatherapi.com/v1/forecast.json?key={}&q={}&days=1&aqi=no&alerts=no",
+            location, api_key);
 
+        let url = Url::parse(&*url)?;
+        // println!("{}", url);
+        let res = reqwest::get(url).await?.json::<serde_json::Value>().await?;
```

Comment:
thats a hint from the reqwest library that you should use the Url struct directly instead of converting a string to a Url.

you should look at `Url::parse_with_params` at the docs for `Url` 
you can extract `http://api.weatherapi.com/v1/forecast.json` to a const and build the Url using `Url` struct, might look nicer (you can also avoid this way common errors like not creating the URL correctly if there are. spaces in the names)